_docs/concepts/encryption.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: guide
+title: StorageOS Docs - Encryption
+anchor: concepts
+module: concepts/encryption
+---
+
+# Encryption
+
+Encryption is enabled on a per volume basis using the
+`storageos.com/encryption` label (for more information see [Encrypted
+Volumes](/docs/operations/encrypted-volumes)).
+
+## Encryption Keys
+
+StorageOS encrypts volumes on disk using AES-256 in XTS-AES mode with 512 bit
+keys as recomended by [ NIST](
+https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38e.pdf
+), with encryption keys being derived using [ HKDF](
+https://eprint.iacr.org/2010/264.pdf ). 
+
+Keys and Initialisation vectors are generated using the [ crypto/rand ]( https://godoc.org/crypto/rand ) package.
+
+| Key                   | Size                 | Usage                                                     |
+| :-------------------- | :------------------- | :-------------------------------------------------------- |
+| Volume                | 512 bits random data | Used by XTS-AES to encrypt and decrypt disk blocks        |
+| User                  | 256 bits random data | Used to encrypt Volume keys                               |
+| Initialisation Vector | 256 bits random data | Used as 'salt' when encrypting volume keys with user keys |
+
+Encryption keys are created per volume. However user keys are created per
+namespace. This means that volumes in the same namespace have their respective
+volume master key encrypted with the same user key but with different
+initialisation vectors.
+
+The components required to derive the encryption key are stored in a Kubernetes
+secret. By default these secrets are stored in the namespace that StorageOS is
+installed into. As Kubernetes secrets are only base64 encoded, it is recommend
+to [encrypt secrets at
+rest.](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) 
+
+## Namespace Specific Encryption Keys
+
+In order to encrypt volumes StorageOS creates a user key and a volume
+initialisation vector. Each namespace has its own user key that is created
+when the namespace is initialized.
+
+For instance, in the example below there are two `ns-key` secrets in the
+`storageos` namespace because a StorageOS volume has been provisioned in the
+`default` and `mongo` namespaces. A `ns-key` is created for a namespace
+regardless of whether an encrypted volume exists in the namespace or not.
+
+```bash
+$ kubectl get secrets -n storageos | grep ns-key
+ns-key.default                                 Opaque                                1      4d
+ns-key.mongo                                   Opaque                                1      5h
+```
+
+## Encypted volume creation
+
+A volume is encrypted with the key K<sub>vol</sub> that is a randomly generated
```

Comment:
I think we can lose the panels with pseudo maths - it's probably a little more detail than we need. Important point is that we encrypt the volume key before storing it, using a key derived from the namespace/vol init vector.